src/full/Agda/TypeChecking/Unquote.hs

```diff
@@ -1041,6 +1044,35 @@ evalTCM v = do
       locallyReduceAllDefs $ checkFunDef NotDelayed i x cs
       primUnitUnit
 
+    tcPragmaForeign :: Text -> Text -> TCM Term
+    tcPragmaForeign backend code = do
+      modifyTC $ \ st -> let preScSt = stPreScopeState st in st
+        { stPreScopeState = preScSt
+          { stPreForeignCode = Map.insertWith
+            (flip (++))
```

Comment:
Here you are appending `(flip (++))`, but below you are prepending `(++)`.
Could you please add comments in each case justifying your choice?

Reply:
I'm appending foreign code to preserve order, because I've thought that some code may be dependent on foreign code blocks introduced earlier?

Reply:
Good, yes.  Please add this comment in the source code.